tests/e2e/actor_features/actor_features_test.go

```diff
@@ -196,6 +210,107 @@ func TestActorFeatures(t *testing.T) {
+
+		time.Sleep(secondsToCheckTimerAndReminderResult * time.Second)
+
+		// Reset reminder
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+		require.NoError(t, err)
+
+		resp, err := utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		require.True(t, countActorAction(resp, actorID, reminderName) == 1)
+	})
+
+	t.Run("Actor reset reminder.", func(t *testing.T) {
+		// Each test needs to have a different actorID
+		actorID := "1001b"
+
+		// Reset reminder
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+		require.NoError(t, err)
+		// Set reminder
+		req := actorReminderOrTimer{
+			Data:    "reminderdata",
+			DueTime: "1s",
+			Period:  "5s",
+		}
+		reqBody, _ := json.Marshal(req)
+		_, err = utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName), reqBody)
+		require.NoError(t, err)
+
+		time.Sleep(3 * time.Second)
+
+		// Reset reminder (before first period trigger)
+		req.DueTime = "20s"
+		reqBody, _ = json.Marshal(req)
+		_, err = utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName), reqBody)
+
+		time.Sleep(10 * time.Second)
+
+		// Delete reminder
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+		require.NoError(t, err)
+
+		resp, err := utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		require.True(t, countActorAction(resp, actorID, reminderName) == 1)
+	})
+
+	t.Run("Actor reminder with deactivate.", func(t *testing.T) {
+		// Each test needs to have a different actorID
+		actorID := "1001c"
+
+		// Reset reminder
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+		require.NoError(t, err)
+		// Set reminder
+		req := actorReminderOrTimer{
+			Data:    "reminderdata",
+			DueTime: "1s",
+			Period:  "1s",
+		}
+		reqBody, _ := json.Marshal(req)
+		_, err = utils.HTTPPost(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName), reqBody)
+		require.NoError(t, err)
+
+		sleepTime := secondsToCheckTimerAndReminderResult / 2 * time.Second
+		time.Sleep(sleepTime)
+
+		resp, err := utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		firstCount := countActorAction(resp, actorID, reminderName)
+		require.GreaterOrEqual(t, firstCount, 1)
+
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorDeleteURLFormat, externalURL, actorID))
+
+		time.Sleep(sleepTime)
+
+		// Reset reminder
+		_, err = utils.HTTPDelete(fmt.Sprintf(actorInvokeURLFormat, externalURL, actorID, "reminders", reminderName))
+		require.NoError(t, err)
+
+		resp, err = utils.HTTPGet(logsURL)
+		require.NoError(t, err)
+		require.True(t, countActorAction(resp, actorID, reminderName) >= firstCount)
+		require.True(t, countActorAction(resp, actorID, reminderName) >= minimumCallsForTimerAndReminderResult)
```

Comment:
with the line 300 deleting the actor, my understanding is that the reminder is also deleted with it. So sleep time should not affect reminder calls. From dapr docs: https://docs.dapr.io/developing-applications/building-blocks/actors/howto-actors/#actor-reminders

```
Reminders are a mechanism to trigger persistent callbacks on an actor at specified times. 
Their functionality is similar to timers. But unlike timers, reminders are triggered under all 
circumstances until the actor explicitly unregisters them or the actor is explicitly deleted. 
```

in this case the reminder should only be called up to 5 times, the initial sleep period line 293.

Also, I did not see any create Actor or delete actor API in http/api.go. Either I missed it or it is not documented in https://docs.dapr.io/reference/api/actors_api/#user-service-code-calling-dapr.

cc @artursouza is my understanding correct?

Reply:
No. The "DELETE" call will deactivate the actor instance on the app instance and has nothing to do with deleting the reminder. The same actor instance can later be activated in another app instance and the reminders persist.

Reply:
I was going for the deactivate actor to reactivated by reminder as persistence. Do we specifically want to have the actor start on a different app? I can add that in addition to this or replace this test with it. What do you think @mukundansundar @artursouza?

Reply:
If you want to verify that the reminder persists, then the right thing is to restart the app. That way, you are guaranteed to have a "clean slate" in memory.

Reply:
> No. The "DELETE" call will deactivate the actor instance on the app instance and has nothing to do with deleting the reminder. The same actor instance can later be activated in another app instance and the reminders persist. 

So is there a way to delete all instances of an actor from the app? Or if not should the documentation be changed to remove the line `the actor is explicitly deleted`.

> Actually, the trigger of a reminder or timer by itself will activate the actor instance.

In this scenario, isn't it the case that a deactivated actor will no longer have timers run? https://docs.dapr.io/developing-applications/building-blocks/actors/howto-actors/#actor-timers

```
All timers are stopped when the actor is deactivated as part of garbage collection. No timer callbacks are invoked after that. Also, the Dapr actors runtime does not retain any information about the timers that were running before deactivation. It is up to the actor to register any timers that it needs when it is reactivated in the future.
```

Does it make sense to have a test testing this also? As in when the app is restarted, the timer should not trigger again? @artursouza

Reply:
Deactivating an actor instance will "remove" the actor from the app instance but not necessarily stop the timer. The next timer trigger will activate the actor again - which should be tested too.

Today, apps don't control activation or deactivation of an actor instance. If the app does not have timers triggering nor reminders nor method invocations, the actor instances will eventually be deactivated - which is another test scenario.

Reply:
BTW, we should not be scope creep and make sure there are additional PRs for tests beyond the scope of what is being worked on.

Reply:
I'll add the test for restarting the app as that's pretty straight-forward and fits within the scope of the PR. Would you also like the timer tests since they're basically just a parody of the reminders?

Reply:
LGTM.